### PR TITLE
[Docs] Restructure of the Host List

### PR DESCRIPTION
**Pull request by ghost**
### Type

- [ ] Bugfix
- [x] Enhancement
- [ ] New feature

### Description of the changes
Updates link in getting started page to reflect doc's new purpose.
Adds info about unsupported platforms.
Adds Lightsail, Hetzner, Contabo, Galaxygate and Ramnode to providers list. 
Rephrases description for hosting providers.
Replaces host list tables with text.
Re-arranges the location of selfhosting info.
Adds some details about the practice of hosting.

Sorry if i made a mistake, this is my first PR.

### CODE REVIEW

File: docs/host-list.rst [hunk 1]
```diff
@@ -19,12 +19,13 @@ Hosting Information
...
-    to properly operate. This means platforms such as Heroku, Pterodactyl,
-    repl.it, Termux and Docker are **NOT** officially supported. LXC are also
-    unsupported. Workarounds for getting Red  running on those platforms are
-    imperfect due to Red's nature. And you will not be able to recieve support
-    if an issue occurs when hosting on any of these platforms.
+    Due to their inability to handle Red's data structure and meet the
+    conditions of being a supported platform; platforms such as Heroku, 
+    Pterodactyl, repl.it, Termux, LXC and alike are **NOT** officially supported. 
+    Docker support found in Cog-Creators GitHub is also a work in progress 
+    and not ready for daily use. Workarounds for getting Red  running on those
```
Comment (bobloy) [inline docs/host-list.rst]:
```suggestion
    and not ready for daily use. Workarounds for getting Red running on those 
```
Extra space.

**Comment by ghost** [comment]
This should be ready for another review

File: docs/host-list.rst [hunk 2]
```diff
@@ -2,90 +2,150 @@
...
 
 .. note::
     This doc is written for the :ref:`hosting section <getting-started-hosting>`
     of the :ref:`getting started <getting-started>` guide. Please take a look
-    if you don't know how to host Red on a VPS.
+    if you don't know how to host Red.
 
-This is a list of the recommended VPS providers.
+
+| For your instance of Red to stay online it needs to be hosted in a system.
```
Comment (Flame442) [inline docs/host-list.rst]:
```suggestion
| For your instance of Red to stay online 24/7, it needs to be hosted on a dedicated system.
```

File: docs/host-list.rst [hunk 3]
```diff
@@ -2,90 +2,150 @@
...
+
+First, we would like to make something clear:
+
+.. warning::
+    Due to their inability to handle Red's data structure and meet the
+    conditions of being a supported platform; platforms such as Heroku, 
+    Pterodactyl, repl.it, Termux, LXC and alike are **NOT** officially supported. 
+    Docker support found in GitHub is also a work in progress and not ready
+    for daily use. Workarounds for getting Red running on those platforms
+    are imperfect due to Red's nature. And one will not be able to recieve
```
Comment (Flame442) [inline docs/host-list.rst]:
```suggestion
    are imperfect due to Red's nature. You will not be able to receive
```

File: docs/host-list.rst [hunk 4]
```diff
@@ -2,90 +2,150 @@
...
+    support if an issue occurs when hosting on any of these platforms.
+
+
+------------------------------------
+Hosting on a VPS or Dedicated Server
+------------------------------------
+
+| One can host Red in a VPS running Linux or Windows. Using a Linux VPS is the
+  recommened option. Dedicated servers also work but are overpowered and cost
```
Comment (Flame442) [inline docs/host-list.rst]:
```suggestion
  recommended option. Dedicated servers also work but are overpowered and cost 
```

File: docs/host-list.rst [hunk 5]
```diff
@@ -2,90 +2,150 @@
...
+
+
+------------------------------------
+Hosting on a VPS or Dedicated Server
+------------------------------------
+
+| One can host Red in a VPS running Linux or Windows. Using a Linux VPS is the
+  recommened option. Dedicated servers also work but are overpowered and cost 
+  ineffective unless one plans to run a very large bot or use their server for 
+  more than than just hosting Red. After setup, Red can be moved to a different
```
Comment (Flame442) [inline docs/host-list.rst]:
```suggestion
  more than just hosting Red. If you have already created an instance, Red can be moved to a different 
```

File: docs/host-list.rst [hunk 6]
```diff
@@ -2,90 +2,150 @@
...
+-------------------
+Choosing a Provider
+-------------------
+
+| The following are some common providers suitable for hosting Red. With
+  each having their pros and cons, this list is mainly intended to act as a
+  starting point. One should conduct their own research and come to
+  a conclusion depending on their needs and budget, taking into account
+  providers not listed here if desired. The key is the provider offering 
+  a OS supported by Red.
```
Comment (Flame442) [inline docs/host-list.rst]:
```suggestion
  an OS supported by Red.
```

File: docs/host-list.rst [hunk 7]
```diff
@@ -2,90 +2,150 @@
...
+
+| `Scaleway <https://www.scaleway.com/>`_ is a VPS and dedicated server
+ provider French in origin with locations in Poland and Netherlands.
+
+| `DigitalOcean <https://www.digitalocean.com/>`_ is a US based cloud services company 
+ with locations available worldwide, the VPS service is provided under the brand name
+ "Droplet".
+
+| `OVH <https://us.ovhcloud.com/vps/>`_ is a company focused on providing hosting
+ and cloud services with locations in the Europe, North America and Asia Pacific.
```
Comment (Flame442) [inline docs/host-list.rst]:
```suggestion
 and cloud services with locations in Europe, North America and Asia Pacific.
```

File: docs/host-list.rst [hunk 8]
```diff
@@ -2,90 +2,150 @@
...
+ Web Services priced lower than their enterprise offerings.
+
+| `Vultr <https://www.vultr.com/>`_ is a US based provider of VPS and dedicated servers
+ with locations available worldwide.
+
+| `Hetzner Online <https://www.hetzner.com/>`_ is a German VPS and dedicated server
+ provider with locations in Germany and Finland.
+
+| `Contabo <https://contabo.com/>`_ is also a German VPS and dedicated server provider
+ with locations in Germany and United States.
```
Comment (Flame442) [inline docs/host-list.rst]:
```suggestion
 with locations in Germany and the United States.
```
I know, english is confusing sometimes.....

File: docs/host-list.rst [hunk 9]
```diff
@@ -2,90 +2,150 @@
...
-| Additionally, new Google Cloud customers get a $300 credit which is valid
-  for 3 months.
+| **Note:** The free tier offered by AWS for EC2 only lasts for 12 months, while
+ Oracle Cloud and Google Cloud offer always free tiers with limited resources.
+
+| Additionally, new Google Cloud customers get a $300 credit which is valid for 3 months.
 
-Other than that... no. There is no good free VPS hoster, outside of
-persuading somebody to host for you, which is incredibly unlikely.
+| Excluding the above, there is no recommended free VPS host. Persuation of
```
Comment (Flame442) [inline docs/host-list.rst]:
```suggestion
| Excluding the above, there is no recommended free VPS host. Persuasion of
```

File: docs/host-list.rst [hunk 10]
```diff
@@ -2,90 +2,150 @@
...
+    for daily use. Workarounds for getting Red running on those platforms
+    are imperfect due to Red's nature. And one will not be able to recieve
+    support if an issue occurs when hosting on any of these platforms.
+
+
+------------------------------------
+Hosting on a VPS or Dedicated Server
+------------------------------------
+
+| One can host Red in a VPS running Linux or Windows. Using a Linux VPS is the
```
Comment (Flame442) [inline docs/host-list.rst]:
```suggestion
| You can host Red in a VPS running Linux or Windows. Using a Linux VPS is the
```
For ""formal"" writing, you don't want to use "you", but for documentation it's fine

File: docs/host-list.rst [hunk 11]
```diff
@@ -2,90 +2,150 @@
...
-+-------------------------------------+-----------------------------------------------------+
 
 ------------
-Self-hosting
+Self Hosting
 ------------
 
-You can always self-host on your own hardware.
-A Raspberry Pi 3 will be more than sufficient for small to medium sized bots.
+| It's possible to self host Red with one's own hardware. A Raspberry Pi 3
```
Comment (Flame442) [inline docs/host-list.rst]:
```suggestion
| It's possible to self host Red with your own hardware. A Raspberry Pi 3 
```

File: docs/host-list.rst [hunk 12]
```diff
@@ -2,90 +2,150 @@
...
 ------------
 
-You can always self-host on your own hardware.
-A Raspberry Pi 3 will be more than sufficient for small to medium sized bots.
+| It's possible to self host Red with one's own hardware. A Raspberry Pi 3 
+  will have enough computing capacity to handle a small to medium sized bot. 
+  One also has the option of building a computer for this purpose or purchasing 
+  a  rack server as any modern hardware should work without issues. But keep in
+  mind this option can leaves one vulnerable to factors outside their control
+  such as power outages.
```
Comment (Flame442) [inline docs/host-list.rst]:
```suggestion
  You can also host on your own computer or rack server. Any modern hardware 
  should work without issues. However, this option leaves you responsible for
  keeping the bot online by paying for electricity costs and dealing with power outages.
```
I rewrote this section a bit, changed the focus to what I think is the major drawbacks to selfhosting. If you want to rewrite it again in a different way, feel free.

File: docs/host-list.rst [hunk 13]
```diff
@@ -2,90 +2,150 @@
...
+
+-------------------
+Choosing a Provider
+-------------------
+
+| The following are some common providers suitable for hosting Red. With
+  each having their pros and cons, this list is mainly intended to act as a
+  starting point. One should conduct their own research and come to
+  a conclusion depending on their needs and budget, taking into account
```
Comment (Flame442) [inline docs/host-list.rst]:
```suggestion
  starting point. You should conduct your own research and come to
  a conclusion depending on your needs and budget, taking into account
```

**Review by Flame442 — CHANGES_REQUESTED** [review]
Mostly some grammar changes. Thank you for your contribution!

File: docs/host-list.rst (hunk 10 above)
Comment (ghost) [reply docs/host-list.rst]: I just tend to use "One" more frequently than most, thank you!

File: docs/host-list.rst (hunk 12 above)
Comment (ghost) [reply docs/host-list.rst]: Better than anything i could come up with

**Review by Flame442 — APPROVED** [review]
All set! Thanks again!

**Comment by ghost** [comment]
Thanks!!

**Pull request merged**